src/Analyzers/VisualBasic/Tests/NewLines/ConsecutiveStatementPlacement/ConsecutiveStatementPlacementTests.vb

```diff
@@ -0,0 +1,375 @@
+            Dim code =
+"
+class C
+    sub M()
+        if (true)
+        end if
+        ' x
+        return
+    end sub
+end class"
+
+            Await TestWithOptionOn(code, code)
+        End Function
+
+        <Fact>
+        Public Async Function TestNotAfterStatementsWithSingleBlankLines() As Task
+            Dim code =
+"
+class C
+    sub M()
+        if (true)
+        end if
+
+        return
+    end sub
+end class"
+
+            Await TestWithOptionOn(code, code)
+        End Function
+
+        <Fact>
+        Public Async Function TestNotAfterStatementsWithSingleBlankLinesWithSpaces() As Task
+            Dim code =
+"
+class C
+    sub M()
+        if (true)
+        end if
+        
+        return
+    end sub
+end class"
+
+            Await TestWithOptionOn(code, code)
+        End Function
+
+        <Fact>
+        Public Async Function TestNotAfterStatementsWithMultipleBlankLines() As Task
+            Dim code =
+"
+class C
+    sub M()
+        if (true)
+        end if
+
+        return
+    end sub
+end class"
+
+            Await TestWithOptionOn(code, code)
+        End Function
+
+        <Fact>
+        Public Async Function TestNotAfterStatementsOnMultipleLinesWithPPDirectiveBetween1() As Task
+            Dim code =
+"
+class C
+    sub M()
+        if (true)
+        end if
+#Region """"
+        return
+    end sub
+#End Region
+end class"
+
+            Await TestWithOptionOn(code, code)
+        End Function
+
+        <Fact>
+        Public Async Function TestNotBetweenBlockAndOuterBlocker() As Task
+            Dim code =
+"
+class C
+    sub M()
+        if (true)
+            if (false)
+            end if
+        end if
+    end sub
+end class"
+
+            Await TestWithOptionOn(code, code)
+        End Function
+
+        <Fact>
+        Public Async Function TestBetweenBlockAndStatement1() As Task
+            Await TestWithOptionOn("
+class C
+    sub M()
+        if (true)
+        [|end if|]
+        return
+    end sub
+end class", "
+class C
+    sub M()
+        if (true)
+        end if
+
+        return
+    end sub
+end class")
+        End Function
+
+        <Fact>
+        Public Async Function TestNotBetweenBlockAndStatement1_WhenOptionOff() As Task
+            Dim code = "
+class C
+    sub M()
+        if (true)
+        end if
+        return
+    end sub
+end class"
+
+            Await New VerifyVB.Test() With {
+                .TestCode = code,
+                .FixedCode = code
+                }.RunAsync()
+        End Function
+
+        <Fact>
+        Public Async Function TestBetweenForEachAndStatement1() As Task
+            For Each x In {0}
+            Next
+
+            Await TestWithOptionOn("
+class C
+    sub M()
+        For Each x In {0}
+        [|Next|]
+        return
+    end sub
+end class", "
+class C
+    sub M()
+        For Each x In {0}
+        Next
+
+        return
+    end sub
+end class")
```

Comment:
Should the following code have the same behavior?

```vb
        Do While True
        Loop
```

Reply:
yup!